feat(web): added edit profile modal

Simplyalex99 opened the pull request:
Added Edit Modal component in Components directory.

Also added root div to render modal in _document.tsx file

Also added state in user/[username]/index.tsx
to detect when modal is clicked or not. Added change from "follow" to "edit profile" and conditional to render modal depending on the state value.

Changed edit.tsx by removing navbar.

Requires css to be converted to tailwindcss in EditModal.tsx

Note: ormconfig.tsx file is not included in push request. In addition, thoughts on implementing custom Modal instead for both edit profile option and send message? Thanks!

Simplyalex99 committed:
Added Edit Modal and changes to two other files

Added Edit Modal component in Components directory.

Also added root div to render modal in _document.tsx file

Also added state in user/[username]/index.tsx
to detect when modal is clicked or not. Added change from "follow" to "edit profile" and conditional to render modal depending on the state value.

Changed edit.tsx by removing navbar.

Requires css to be converted to tailwindcss in EditModal.tsx

heybereket committed:
Merge branch 'staging' into staging

HenryLeC commented on packages/web/src/components/EditModal.tsx:
```diff
@@ -0,0 +1,91 @@
+};
+//  '#0c111b' style background  color for modal
+const MODAL_STYLES: CSS.Properties = {
+  position: 'absolute',
+  backgroundColor: '#0c111b',
+  zIndex: 1000,
+  borderRadius: '25px',
+  gridArea: 'modal',
+  left: '50%',
+  width: '50%',
+
+  top: '50%',
+  transform: 'translate(-50%, -50%)',
+  transition: 'all ease-in .4s'
+
+};
+
+const OUTER_STYLE: CSS.Properties = {
+  position: 'fixed',
+  top: 0,
+  left: 0,
+  bottom: 0,
+  right: 0,
+  backgroundColor: 'rgba(0, 0, 0, .7)',
+  zIndex: 1000,
+  overflow: 'hidden',
+
+};
+
+const CLOSE_BUTTON_STYLE: CSS.Properties = {
+  backgroundColor: '#232936',
+  borderRadius: '50px',
+  outline: 'none',
+  zIndex: 1000,
+  display: 'inline-block',
+  transform: 'translate(-100%, -10%)',
+
+
+
+};
```
Can you change this to tailwind before merge please

Simplyalex99 replied:
Hello, I will try , Acronie was planning on doing the tailwind changes but il give it a go.

Simplyalex99 commented:
Hello,  i forgot to add one comment about the typeof window==undefined in EditModal.tsx. This was to prevent nextjs from crashing sometimes because it  might check server side first and DOM is undefined at that point.

HenryLeC committed:
feat: Add ability to like and dislike posts

HenryLeC committed:
Merge branch 'staging' of https://github.com/oasis-sh/oasis into staging

Simplyalex99 committed:
Merge branch 'oasis-sh:staging' into staging

BronzW commented on packages/api/schema.gql:
```diff
@@ -64,6 +64,7 @@ type Mutation {
   editPost(data: EditPostInput!, postId: String!): Boolean!
   giveBadge(badgeName: String!, username: String!): Boolean!
   joinResort(resortId: String!): Boolean
+  likeDislike(dislike: Boolean!, like: Boolean!, postId: String!): Boolean!
   makeAdmin(roles: [Role!]!, user: String!): Boolean!
   makeBadge(data: MakeBadgeInput!): Boolean!
   markNotificationAsRead(notificationId: String!): Boolean!
```
why is this being changed

BronzW replied:
wait nvm

Simplyalex99 committed:
Added re-usable Modal

Called Modal Component in username/index.tsx file

Changed old modal version EditModal to pure tailwind css.
